icu4c/source/test/fuzzer/collator_compare_fuzzer.cpp

```diff
@@ -10,23 +11,36 @@
 
 IcuEnvironment* env = new IcuEnvironment();
 
+static const std::array<icu_74::Collator::ECollationStrength, 5> kStrength = {
+    icu::Collator::PRIMARY,
+    icu::Collator::SECONDARY,
+    icu::Collator::TERTIARY,
+    icu::Collator::QUATERNARY,
+    icu::Collator::IDENTICAL
+};
+
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
   UErrorCode status = U_ZERO_ERROR;
 
   if (size < 2)
     return 0;
 
+  uint16_t rnd = *(reinterpret_cast<const uint16_t *>(data));
```

Comment:
You are now using the first two data bytes for the locale, the strength, and the first char16_t of the compbuff1 string.

Wouldn't it improve the fuzzer if these used separate parts of the data? For example,
- locale = bytes 0..1
- strength = byte 2
- compbuff1 = first half of remaining bytes
- compbuff2 = second half of remaining bytes

It seems easy to do so: On line 25 require at least 5 bytes, and when you read the locale value and the strength, increase the data pointer and decrease the size.

Reply:
ok, I shift the data by two bytes now. We really do not need to have separate bytes for locale and strength because random 16 bits number is much greater than the permutation of strength and number of locale. Just a rnd16 %5 and rnd16/5 is good enough.